pkg/controllers/awsloadbalancercontroller/rbac_clusterrole.go

```diff
@@ -0,0 +1,131 @@
+package awsloadbalancercontroller
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	albo "github.com/openshift/aws-load-balancer-operator/api/v1alpha1"
+)
+
+// ensureClusterRoles creates and maintains clusterroles required by the aws-load-balancer controller.
+func (r *AWSLoadBalancerControllerReconciler) ensureClusterRoles(ctx context.Context, controller *albo.AWSLoadBalancerController) error {
+	reqLogger := log.FromContext(ctx)
+
+	desired := desiredClusterRoles(ctx)
+
+	for _, obj := range desired {
+		if err := controllerutil.SetControllerReference(controller, obj, r.Scheme); err != nil {
+			return fmt.Errorf("failed to set the controller reference for clusterroles %s : %w", obj.Name, err)
+		}
+	}
+
+	exist, current, err := r.currentClusterRoles(ctx, desired)
+	if err != nil {
+		return fmt.Errorf("failed to fetch current clusterroles: %w", err)
+	}
+
+	if !exist {
+		if err := r.createClusterRoles(ctx, desired); err != nil {
+			return err
+		}
+		_, current, err = r.currentClusterRoles(ctx, desired)
```

Comment:
You don't need to get the cluster role again after creating it because it's not passed back to the main controller reconciler loop.

Reply:
Addressed